src/plugins/safeSnap/components/HandleOutcome.vue

```diff
@@ -345,22 +346,41 @@ const questionState = computed(() => {
     return QuestionStates.questionNotSet;
 
   const ts = (Date.now() / 1e3).toFixed();
-  const { finalizedAt, cooldown, expiration, executionApproved, nextTxIndex } =
-    questionDetails.value;
+  const {
+    finalizedAt,
+    cooldown,
+    expiration,
+    executionApproved,
+    nextTxIndex,
+    proposalEvent
+  } = questionDetails.value;
+
+  // Proposal is approved if it expires without a dispute.
+  if (proposalEvent.isExpired) return QuestionStates.proposalApproved;
+
+  // Proposal is approved if it has been settled without a disputer.
+  if (proposalEvent.isSettled && !proposalEvent.isDisputed)
+    return QuestionStates.proposalApproved;
 
-  const isExpired = finalizedAt + expiration < ts;
+  // TODO: Deleting disputed proposals, deleting resolved proposals that have been executed,
+  // allowing re-proposal of disputed proposals.
 
-  if (!finalizedAt) return QuestionStates.questionNotResolved;
-  if (executionApproved) {
-    if (finalizedAt + cooldown > ts) return QuestionStates.waitingForCooldown;
+  // Commenting out the below since it is not relevant for uma execution
 
-    if (!Number.isInteger(nextTxIndex))
-      return QuestionStates.completelyExecuted;
-    else if (isExpired) return QuestionStates.timeExpired;
+  // const isExpired = finalizedAt + expiration < ts;
 
-    return QuestionStates.proposalApproved;
-  }
-  if (isExpired) return QuestionStates.proposalRejected;
+  // if (!finalizedAt) return QuestionStates.questionNotResolved;
+  // if (executionApproved) {
+  //   if (finalizedAt + cooldown > ts) return QuestionStates.waitingForCooldown;
+
+  //   if (!Number.isInteger(nextTxIndex))
+  //     return QuestionStates.completelyExecuted;
+  //   else if (isExpired) return QuestionStates.timeExpired;
+
+  //   return QuestionStates.proposalApproved;
+  // }
+
+  // if (isExpired) return QuestionStates.proposalRejected;
```

Comment:
Is there a point to leaving this here if we're completely dropping other parts of the reality.eth implementation?

Reply:
Yes, this should be removed. John is currently working on the `HandleOutcome` file and our next PR will remove a lot of the unnecessary parts of the reality.eth implementation.